src/Raven.Studio/typescript/viewmodels/database/indexes/deleteIndexesConfirm.ts

```diff
@@ -3,35 +3,85 @@ import dialog = require("plugins/dialog");
 import database = require("models/resources/database");
 import dialogViewModelBase = require("viewmodels/dialogViewModelBase");
 import messagePublisher = require("common/messagePublisher");
+import index = require("models/database/index/index");
+import indexDefinition = require("models/database/index/indexDefinition");
 
-class deleteIndexesConfirm extends dialogViewModelBase {
+class indexInfoForDelete {
+    indexName: string;
+    reduceOutputCollection: string;
+    ReferenceCollection: string;
 
-    deleteTask = $.Deferred<boolean>();
-    title: string;
+    readonly referenceCollectionExtension = "/References";
+    
+    constructor(name: string, reduceCollection: string, hasReferenceCollection: boolean) {
+        this.indexName = name;
+        this.reduceOutputCollection = reduceCollection;
+        this.ReferenceCollection = hasReferenceCollection ? this.reduceOutputCollection + this.referenceCollectionExtension : "";
+    }
+}
 
-    constructor(private indexNames: string[], private db: database, title?: string) {
+class deleteIndexesConfirm extends dialogViewModelBase {
+    title: string;
+    subTitleHtml: string;
+    warning: KnockoutComputed<string>;
+    deleteTask = $.Deferred<boolean>();
+    
+    indexesInfoForDelete = ko.observableArray<indexInfoForDelete>();
+    
+    constructor(private indexes: index[] | indexDefinition[], private db: database) {
         super();
+        this.initObservables();
 
-        if (!indexNames || indexNames.length === 0) {
+        if (!indexes || indexes.length === 0) {
             throw new Error("Indexes must not be null or empty.");
         }
-
-        this.title = title || (indexNames.length === 1 ? "Delete index?" : "Delete indexes?");
+        
+        if (indexes[0] instanceof index) {
+            // Coming here from Indexes List View
+            this.indexesInfoForDelete((indexes as index[]).map(x => {
```

Comment:
use factory here. No need for { return ...} when single statement. You simply write:
map(x => indexInfoForDelete.fromIndex(x))

Reply:
done